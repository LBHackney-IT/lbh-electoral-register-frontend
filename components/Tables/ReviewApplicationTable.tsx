import { Elector } from 'types';

export interface ReviewApplicationTableProps {
  elector: Elector;
}

export interface TableRowProps {
  name: string;
  content: (string | undefined)[];
}

const TableRow = ({ name, content }: TableRowProps): React.ReactElement => (
  <div className="govuk-summary-list__row">
    <dt className="govuk-summary-list__key half-width">{name}</dt>
    <dd className="govuk-summary-list__value half-width">
      {content.map((row, index) => (
        <p key={`${name}-${index}`} className="govuk-body">
          {row}
        </p>
      ))}
    </dd>
  </div>
);

const ReviewApplicationTable = ({
  elector,
}: ReviewApplicationTableProps): React.ReactElement => (
  <div>
    <dl className="govuk-summary-list">
      <TableRow
        name="Name"
        content={[`${elector.first_name} ${elector.last_name}`]}
      />
      <TableRow
        name="Address"
        content={[elector.address_ln1, elector.address_ln2, elector.postcode]}
      />
      <TableRow
        name="Lives at another address?"
        content={[elector.lived_at_another_address === 'Y' ? 'Yes' : 'No']}
      />
      <TableRow
        name="Moved house in the last 12 months?"
        content={[elector.moved_house === 'Y' ? 'Yes' : 'No']}
      />

      <TableRow
        name="Previous Address"
        content={[
          elector.previous_address_ln1,
          elector.previous_address_ln2,
          elector.previous_postcode,
          elector.previous_address_ln1
            ? `Overseas voter? ${
                elector.previous_address_overseas === 'Y' ? ' Yes' : ' No'
              }`
            : '',
        ]}
      />

      <TableRow
        name="Contact details"
        content={[elector.email_address, elector.phone_input]}
      />

      <TableRow name="Nationality" content={[elector.nationality]} />

      <TableRow
        name="Date of birth"
        content={[elector.dob || elector.age_range]}
      />

      <TableRow name="National insurance number" content={[elector.nino]} />

      <TableRow
        name="Reasons for no nationality, date of birth, or national insurance
          number"
        content={[elector.reasons]}
      />

      <TableRow
        name="Previous name"
        content={[
          elector.previous_name,
          elector.previous_name_date
            ? `Date changed: ${elector.previous_name_date}`
            : '',
        ]}
      />

      <TableRow
        name="Voting by post or proxy?"
        content={[
          elector.voting_proxy_type
            ? elector.voting_proxy_type === 'post'
              ? 'Post'
              : 'Proxy'
            : '',
        ]}
      />

      {elector.open_register && (
        <TableRow
          name="Do not want to be included in the open register?"
          content={[elector.open_register ? 'True' : 'False']}
        />
      )}

      <TableRow
        name="Applicant signature provided?"
        content={[
          'Yes',
          elector.signature_date
            ? `Date signed: ${elector.signature_date}`
            : '',
        ]}
      />
    </dl>
  </div>
);

export default ReviewApplicationTable;
