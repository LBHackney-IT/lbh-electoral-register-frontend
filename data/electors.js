export default [
  {
    address_ln1: '1 Hillman Street',
    address_ln2: 'London, United Kingdom',
    changed_name: true,
    declaration: true,
    dob: '1990-07-04',
    email_address: 'fiona.sampley@gmail.com',
    first_name: 'Fiona',
    last_name: 'Sampley',
    lived_at_another_address: 'N',
    moved_house: 'N',
    nationality: 'British',
    nino: 'QQ123456A',
    open_register: true,
    phone_input: '1234567890',
    postcode: 'E8 1DY',
    previous_name: 'Fiona Exampley',
    previous_name_date: '2012-12-12',
    reasons: '',
    signature_date: '2021-05-04',
    time_created: '2021-06-22 11:07:09',
    urn: 'cdaa2-b760-40a0-5205-5a2e40007c0',
    voting_proxy_check: false,
    verification_date: '2021-05-14',
  },
  {
    address_ln1: '5 Morning Lane',
    address_ln2: 'London, United Kingdom',
    changed_name: false,
    declaration: true,
    dob: '1984-10-31',
    email_address: 'hassan@example.com',
    first_name: 'Hassan',
    last_name: 'Exampleson',
    lived_at_another_address: 'Y',
    moved_house: 'Y',
    nationality: 'British',
    nino: 'OK543210Y',
    open_register: false,
    phone_input: '07333222111',
    postcode: 'E9 6ND',
    previous_address_ln1: '24 Chestnut Avenue',
    previous_address_ln2: 'London, United Kingdom',
    previous_address_overseas: 'N',
    previous_postcode: 'N8 8NY',
    reasons: '',
    signature_date: '2021-06-14',
    time_created: '2021-06-22 11:40:11',
    urn: '0816b-7b44-2f87-10bd-a05f7adb36a',
    voting_proxy_check: true,
    voting_proxy_type: 'post',
    verification_date: '2021-06-21',
  },
  {
    address_ln1: '224 Homerton High Street',
    address_ln2: 'London, United Kingdom',
    changed_name: false,
    declaration: true,
    dob: '1960-12-25',
    email_address: 'brittany.faker@yahoo.com',
    first_name: 'Brittany',
    last_name: 'Faker',
    lived_at_another_address: 'N',
    moved_house: 'N',
    nationality: 'British',
    nino: 'ME163599B',
    open_register: false,
    phone_input: '071099177600',
    postcode: 'E9 6AS',
    reasons: '',
    signature_date: '2021-06-21',
    time_created: '2021-06-22 11:48:54',
    urn: 'af884-e15b-415-c84-b5ce0e7d8a4',
    voting_proxy_check: false,
    verification_date: '2021-06-25',
  },
];
