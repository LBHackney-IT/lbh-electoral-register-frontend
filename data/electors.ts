import { Elector } from 'types';

export const electors: Elector[] = [
  {
    address_ln1: '1 Hillman Street',
    address_ln2: 'London, United Kingdom',
    changed_name: true,
    declaration: true,
    dob: '1990-07-04',
    email_address: 'fiona.sampley@gmail.com',
    first_name: 'Fiona',
    last_name: 'Sampley',
    lived_at_another_address: 'N',
    moved_house: 'N',
    nationality: 'British',
    nino: 'QQ123456A',
    open_register: true,
    phone_input: '1234567890',
    postcode: 'E8 1DY',
    previous_name: 'Fiona Exampley',
    previous_name_date: '2012-12-12',
    reasons: '',
    signature_date: '2021-05-04',
    time_created: '2021-06-22 11:07:09',
    urn: 'cdaa2-b760-40a0-5205-5a2e40007c0',
    voting_proxy_check: false,
    verification_date: '2021-05-14',
  },
  {
    address_ln1: '5 Morning Lane',
    address_ln2: 'London, United Kingdom',
    changed_name: false,
    declaration: true,
    dob: '1984-10-31',
    email_address: 'hassan@example.com',
    first_name: 'Hassan',
    last_name: 'Exampleson',
    lived_at_another_address: 'Y',
    moved_house: 'Y',
    nationality: 'British',
    nino: 'OK543210Y',
    open_register: false,
    phone_input: '07333222111',
    postcode: 'E9 6ND',
    previous_address_ln1: '24 Chestnut Avenue',
    previous_address_ln2: 'London, United Kingdom',
    previous_address_overseas: 'N',
    previous_postcode: 'N8 8NY',
    reasons: '',
    signature_date: '2021-06-14',
    time_created: '2021-06-22 11:40:11',
    urn: '0816b-7b44-2f87-10bd-a05f7adb36a',
    voting_proxy_check: true,
    voting_proxy_type: 'post',
    verification_date: '2021-06-21',
  },
  {
    address_ln1: '224 Homerton High Street',
    address_ln2: 'London, United Kingdom',
    changed_name: false,
    declaration: true,
    dob: '1960-12-25',
    email_address: 'brittany.faker@yahoo.com',
    first_name: 'Brittany',
    last_name: 'Faker',
    lived_at_another_address: 'N',
    moved_house: 'N',
    nationality: 'British',
    nino: 'ME163599B',
    open_register: false,
    phone_input: '071099177600',
    postcode: 'E9 6AS',
    reasons: '',
    signature_date: '2021-06-21',
    time_created: '2021-06-22 11:48:54',
    urn: 'af884-e15b-415-c84-b5ce0e7d8a4',
    voting_proxy_check: false,
    verification_date: '2021-06-25',
  },
];

<<<<<<< HEAD
export const headers = [
  { label: 'Address line 1', key: 'address_ln1' },
  { label: 'Address line 2', key: 'address_ln2' },
  { label: 'Changed name?', key: 'changed_name' },
  { label: 'Age range', key: 'age_range' },
  { label: 'Singature?', key: 'declaration' },
  { label: 'Date of birth', key: 'dob' },
  { label: 'Email address', key: 'email_address' },
  { label: 'First name', key: 'first_name' },
  { label: 'Last name', key: 'last_name' },
  { label: 'Lived at another address?', key: 'lived_at_another_address' },
  { label: 'Moved house in the last 12 months?', key: 'moved_house' },
  { label: 'Nationality', key: 'nationality' },
  { label: 'National insurance number', key: 'nino' },
  {
    label: 'Do not want to be included in the open register?',
    key: 'open_register',
  },
  { label: 'Phone number', key: 'phone_input' },
  { label: 'Postcode', key: 'postcode' },
  { label: 'Previous address line 1', key: 'previous_address_ln1' },
  { label: 'Previous address line 2', key: 'previous_address_ln2' },
  { label: 'Previous address overseas?', key: 'previous_address_overseas' },
  { label: 'Previous name', key: 'previous_name' },
  { label: 'Date of name change', key: 'previous_name_date' },
  { label: 'Previous postcode', key: 'previous_postcode' },
  {
    label:
      'Reasons for no nationality, date of birth, or national insurance number',
    key: 'reasons',
  },
  { label: 'Date of signature', key: 'signature_date' },
  { label: 'Time created', key: 'time_created' },
  { label: 'Unique ID', key: 'urn' },
  { label: 'Verification date', key: 'verification_date' },
  { label: 'Voting by post or proxy?', key: 'voting_proxy_check' },
  { label: 'Voting proxy type', key: 'voting_proxy_type' },
];
=======
export const testElector: Elector = {
  address_ln1: 'Test address_ln1',
  address_ln2: 'Test address_ln2',
  age_range: 'Test age_range',
  changed_name: true,
  declaration: true,
  dob: 'Test dob',
  email_address: 'Test email_address',
  first_name: 'Test first_name',
  last_name: 'Test last_name',
  lived_at_another_address: 'Test lived_at_another_address',
  moved_house: 'Test moved_house',
  nationality: 'Test nationality',
  nino: 'Test nino',
  open_register: true,
  phone_input: 'Test phone_input',
  postcode: 'Test postcode',
  previous_address_ln1: 'Test previous_address_ln1',
  previous_address_ln2: 'Test previous_address_ln2',
  previous_address_overseas: 'Test previous_address_overseas',
  previous_name: 'Test previous_name',
  previous_name_date: 'Test previous_name_date',
  previous_postcode: 'Test previous_postcode',
  reasons: 'Test reasons',
  signature_date: 'Test signature_date',
  time_created: 'Test time_created',
  urn: 'Test-urn',
  verification_date: 'Test verification_date',
  voting_proxy_check: true,
  voting_proxy_type: 'Test voting_proxy_type',
};
>>>>>>> 512281c (created test elector for testing purposes)
