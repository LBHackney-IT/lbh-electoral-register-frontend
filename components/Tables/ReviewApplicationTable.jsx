import { electorPropTypes } from '../../data/electors';

const ReviewApplicationTable = ({ elector }) => (
  <div>
    <dl className="govuk-summary-list">
      <div className="govuk-summary-list__row">
        <dt className="govuk-summary-list__key">Name</dt>
        <dd className="govuk-summary-list__value">
          {elector.first_name} {elector.last_name}
        </dd>
      </div>
      <div className="govuk-summary-list__row">
        <dt className="govuk-summary-list__key">Address</dt>
        <dd className="govuk-summary-list__value">
          <p className="govuk-body">{elector.address_ln1}</p>
          <p className="govuk-body">
            {elector.address_ln2} {elector.postcode}
          </p>
        </dd>
      </div>
      <div className="govuk-summary-list__row">
        <dt className="govuk-summary-list__key">Lives at another address?</dt>
        <dd className="govuk-summary-list__value">
          <p className="govuk-body">
            {elector.lived_at_another_address === 'Y' ? 'Yes' : 'No'}
          </p>
        </dd>
      </div>
      <div className="govuk-summary-list__row">
        <dt className="govuk-summary-list__key">
          Moved house in the last 12 months?
        </dt>
        <dd className="govuk-summary-list__value">
          <p className="govuk-body">
            {elector.moved_house === 'Y' ? 'Yes' : 'No'}
          </p>
        </dd>
      </div>

      {elector.previous_address_ln1 && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">Previous Address</dt>
          <dd className="govuk-summary-list__value">
            <div>
              <p className="govuk-body">{elector.previous_address_ln1}</p>
              <p className="govuk-body">
                {elector.previous_address_ln2} {elector.previous_postcode}
              </p>
              <p className="govuk-body">
                Overseas voter?
                {elector.previous_address_overseas === 'Y' ? ' Yes' : ' No'}
              </p>
            </div>
          </dd>
        </div>
      )}

      {(elector.email_address || elector.phone_input) && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">Contact details</dt>
          <dd className="govuk-summary-list__value">
            <p className="govuk-body">{elector.email_address}</p>
            <p className="govuk-body">{elector.phone_input}</p>
          </dd>
        </div>
      )}

      {elector.nationality && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">Nationality</dt>
          <dd className="govuk-summary-list__value">{elector.nationality}</dd>
        </div>
      )}

      {elector.dob && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">Date of birth</dt>
          <dd className="govuk-summary-list__value">
            {elector.dob}
            {elector.age_range}
          </dd>
        </div>
      )}

      {elector.nino && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">National insurance number</dt>
          <dd className="govuk-summary-list__value">{elector.nino}</dd>
        </div>
      )}

      {elector.reasons && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">
            Reasons for no nationality, date of birth, or national insurance
            number
          </dt>
          <dd className="govuk-summary-list__value">{elector.reasons}</dd>
        </div>
      )}

      {elector.changed_name && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">Previous name</dt>
          <dd className="govuk-summary-list__value">
            <div>
              <p className="govuk-body">{elector.previous_name}</p>
              <p className="govuk-body">
                Date changed: {elector.previous_name_date}
              </p>
            </div>
          </dd>
        </div>
      )}

      {elector.voting_proxy_check && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">Voting by post or proxy?</dt>
          <dd className="govuk-summary-list__value">
            <p className="govuk-body">{elector.voting_proxy_type}</p>
          </dd>
        </div>
      )}

      {elector.open_register && (
        <div className="govuk-summary-list__row">
          <dt className="govuk-summary-list__key">
            Do not want to be included in the open register?
          </dt>
          <dd className="govuk-summary-list__value">
            {elector.open_register ? 'True' : 'False'}
          </dd>
        </div>
      )}

      <div className="govuk-summary-list__row">
        <dt className="govuk-summary-list__key">
          Applicant signature provided?
        </dt>
        <dd className="govuk-summary-list__value">
          <p className="govuk-body">Yes</p>
          {elector.signature_date && (
            <p className="govuk-body">Date signed: {elector.signature_date}</p>
          )}
        </dd>
      </div>
    </dl>
  </div>
);

ReviewApplicationTable.propTypes = {
  elector: {
    ...electorPropTypes,
  },
};

export default ReviewApplicationTable;
