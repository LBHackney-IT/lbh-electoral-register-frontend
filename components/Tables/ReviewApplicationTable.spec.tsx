import { render } from '@testing-library/react';
import ReviewApplicationTable from './ReviewApplicationTable';
import { testElector } from 'data/electors';

describe('ReviewApplicationTable', () => {
  it('should render elector data properly', () => {
    const { asFragment } = render(
      <ReviewApplicationTable elector={testElector} />
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
