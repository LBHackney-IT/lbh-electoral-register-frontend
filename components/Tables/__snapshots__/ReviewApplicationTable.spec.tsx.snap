// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewApplicationTable should render elector data properly 1`] = `
<DocumentFragment>
  <div>
    <dl
      class="govuk-summary-list"
    >
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Name
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test first_name Test last_name
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Address
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test address_ln1
          </p>
          <p
            class="govuk-body"
          >
            Test address_ln2
          </p>
          <p
            class="govuk-body"
          >
            Test postcode
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Lives at another address?
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            No
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Moved house in the last 12 months?
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            No
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Previous Address
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test previous_address_ln1
          </p>
          <p
            class="govuk-body"
          >
            Test previous_address_ln2
          </p>
          <p
            class="govuk-body"
          >
            Test previous_postcode
          </p>
          <p
            class="govuk-body"
          >
            Overseas voter?  No
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Contact details
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test email_address
          </p>
          <p
            class="govuk-body"
          >
            Test phone_input
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Nationality
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test nationality
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Date of birth
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test dob
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          National insurance number
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test nino
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Reasons for no nationality, date of birth, or national insurance number
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test reasons
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Previous name
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Test previous_name
          </p>
          <p
            class="govuk-body"
          >
            Date changed: Test previous_name_date
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Voting by post or proxy?
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Proxy
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Do not want to be included in the open register?
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            True
          </p>
        </dd>
      </div>
      <div
        class="govuk-summary-list__row"
      >
        <dt
          class="govuk-summary-list__key half-width"
        >
          Applicant signature provided?
        </dt>
        <dd
          class="govuk-summary-list__value half-width"
        >
          <p
            class="govuk-body"
          >
            Yes
          </p>
          <p
            class="govuk-body"
          >
            Date signed: Test signature_date
          </p>
        </dd>
      </div>
    </dl>
  </div>
</DocumentFragment>
`;
